src/lib/forms/core/SelectField.js

```diff
@@ -82,7 +81,7 @@ export class SelectField extends Component {
       options,
       ...uiProps
     } = this.props;
-    const value = getIn(values, fieldPath, multiple ? [] : defaultValue);
+    const value = getIn(values, fieldPath, multiple);
```

Comment:
My mistake in suggesting you this change, it looks like value is used here, we might want to simply do:
```
     const value = getIn(values, fieldPath, multiple ? [] : ''); 
```
The other changes look ok to me